File: Machine Learning A-Z - udemy - superdatascience/import.R
# data prep

# import
dataset <- read.csv("Data.csv")
dataset

# missing data
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN=function(x) mean(x, na.rm=TRUE)),
                     dataset$Age)

dataset$Salary = ifelse(is.na(dataset$Salary),
                     ave(dataset$Salary, FUN=function(x) mean(x, na.rm=TRUE)),
                     dataset$Salary)

# ENCODING CATEGORICAL AND DUMMY VARIABLE
dataset$Country = factor(dataset$Country,
                         levels = c('France', 'Spain', 'Germany'),
                         labels = c(1,2,3))

dataset$Purchased = factor(dataset$Purchased,
                         levels = c('No', 'Yes'),
                         labels = c(0,1))

#SPLIT TRAIN AND TEST
#install.packages('caTools')
library(caTools)
# set.seed(123) if we would like to see the same result
set.seed(123)
split=sample.split(dataset$Purchased, SplitRatio = 0.8)
train_set = subset(dataset, split==TRUE)
test_set = subset(dataset, split==FALSE)


# FEATURE SCALING
# scale only the numeric columns and exclude the factor columns
train_set[,2:3] = scale(train_set[,2:3])
test_set[,2:3] = scale(test_set[,2:3])

File: Machine Learning A-Z - udemy - superdatascience/import.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jul 23 11:12:00 2018

@author: Polla
"""

import pandas as pd

dataset=pd.read_csv('Data.csv')
print(dataset)
X=dataset.iloc[:,:-1].values
print("X values are:")
print(X)
y=dataset.iloc[:,3].values
print("y values are:")
print(y)

# MIMSSING DATA
from sklearn.preprocessing import Imputer
imputer  = Imputer(missing_values='NaN', strategy= 'mean', axis=0)
imputer.fit(X[:,1:3])
X[:, 1:3]=imputer.transform(X[:, 1:3])
print("X values are after the imputer")
print(X)

#CATEGORICAL AND DUMMY VARIABLE
# Categorical variable transform to integer
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X=LabelEncoder()
# country column
X[:,0] = labelencoder_X.fit_transform(X[:,0])
print("X values are after the first labelencoder")
print(X)
# create dummy variables
onehotencoder=OneHotEncoder(categorical_features=[0])
X = onehotencoder.fit_transform(X).toarray()
print("X values are after the first dummy transform")
print(X)

labelencoder_y=LabelEncoder()
y = labelencoder_y.fit_transform(y)


# SPLITTING TRAIN AND TEST SET
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.2, random_state=0)
#random_state=0 if we would like to see the same result


# FEATURE SCALING
from sklearn.preprocessing import StandardScaler
sc_X=StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)

